### ISSUE TITLE
IPAD print Issue in chrome 

### ISSUE THREAD

**Issue by shaangidwani** [crabbly/Print.js#328]
while print in ipad chrome it will print all the html page with background instead of pdf or select particular form.

kindly help in this matter asap.

**Comment by crabbly** [comment]
Hey @shaangidwani, looks like a browser compatibility issue, with Chrome in IOS.
This has already been reported in #312.

I currently don't know on how we are going to fix this. Any ideas and feedback are welcomed. 
Thank you.

**Comment by songlim327** [comment]
Hi @crabbly, just a suggestion from what I had did in my project. Instead of printing a file from an iframe, I had open it in a new tab with `FileReader`. In there, users may proceed with the printing job just like Firefox in Windows.

**Comment by crabbly** [comment]
I agree. We need to add this check for Chrome in IOS. 

Until we improve support for this browsers, the target pdf should always open in a new tab to prevent errors.

The first thing we need to do is to create a method to our our [`brower.js`](https://github.com/crabbly/Print.js/blob/master/src/js/browser.js) file that identifies Chrome in IOS.

**Comment by songlim327** [comment]
I had forked a copy of Print.js and created a commit which add a method to identify Chrome in IOS. Should I submit a pull request?

**Comment by crabbly** [comment]
Hey @songlim327,
Yes, please do. That would be great. Any help is appreciated.
Thank you!

**Comment by Chaitanya21** [comment]
Is there a fix for this yet?   The goal of using PrintJS was to be able to open the print dialog box and immediately print without the need for a new tab.    So this issue breaks that feature with iOS and Chrome.  Anything that I might be able to do to help get it resolved?

**Comment by shaangidwani** [comment]
anyone got anything please help.

**Comment by bartolomeukuma** [comment]
any solution?

**Comment by Chaitanya21** [comment]
Nope.   Never got a solution here.   We even ended up getting a hold of Apple as older versions were working properly, but even so, never got a solution.

**Comment by evanb2** [comment]
I wonder if this is related:
https://github.com/jasonday/printThis/issues/152#issuecomment-511628889

**Comment by timoostrich** [comment]
Looks like this is still a problem on iOS/iPadOS Chrome and Firefox (Safari is fine).
Any plans to fix it? Or are there other modern solutions?

**Comment by uavn** [comment]
Workaround, that I used:

```js
function printJSFixed(options) {
    if (-1 !== navigator.userAgent.toLowerCase().indexOf('crios') && 'pdf' === options.type && options.base64) {
        var newWindow = window.open('', '_blank');
        newWindow.document.write('<embed width="100%" height="100%" src="data:application/pdf;base64,' + options.printable + '" type="application/pdf"/>');
        newWindow.print();
    } else {
        printJS(options);
    }
}
```

and replace your call:
```js
printJS({printable: pdf, type: 'pdf', base64: true});
```

to
```js
printJSFixed({printable: pdf, type: 'pdf', base64: true});
```

for Chrome on iOS - it opens new tab, renders PDF in it, and calls print dialog, for the rest - print-js will be used like before.

**Comment by algarsys** [comment]
Fix for printJS html: 
function printJSFixed(options) {
        if (-1 !== navigator.userAgent.toLowerCase().indexOf('crios') || -1 !== navigator.userAgent.toLowerCase().indexOf('fxios')) {        var newWindow = window.open('', '_blank');
        newWindow.document.write($('#'+options).html());
        newWindow.print();
    } else {
        printJS(options,'html');
    }
}
